web/pandas/pdeps/0005-no-default-index-mode.md

```diff
@@ -0,0 +1,318 @@
+  ``Index``.
+
+However, neither of the above will be discussed here.
+
+## Detailed Description
+
+The core pandas code would change as little as possible. The additional complexity should be handled
+within the ``NoRowIndex`` object. It would act just like ``RangeIndex``, but would be a bit stricter
+in some cases:
+- `name` could only be `None`;
+- `start` could only be `0`, `step` `1`;
+- when appending a ``NoRowIndex``, the result would still be ``NoRowIndex``;
+- the ``NoRowIndex`` class would be preserved under slicing;
+- it could only be aligned with another ``Index`` if it's also ``NoRowIndex`` and if it's of the same length;
+- ``DataFrame`` columns can't be `NoRowIndex` (so ``transpose`` would need some adjustments when called on a ``NoRowIndex`` ``DataFrame``);
+- `insert` and `delete` should raise. As a consequence, if ``df`` is a ``DataFrame`` with a
+  ``NoRowIndex``, then `df.drop` with `axis=0` would always raise;
+- arithmetic operations (e.g. `NoRowIndex(3) + 2`) would always raise.
+- when printing a ``DataFrame``/``Series`` with a ``NoRowIndex``, then the row labels wouldn't be printed.
+
+Let's go into more detail for some of these.
+
+### NoRowIndex.append
+
+If one has two ``DataFrame``s with ``NoRowIndex``, then one would expect that concatenating them would
+result in a ``DataFrame`` which still has ``NoRowIndex``. To do this, the following rule could be introduced:
+
+> If appending a ``NoRowIndex`` of length ``y`` to a ``NoRowIndex`` of length ``x``, the result will be a
+  ``NoRowIndex`` of length ``x + y``.
+
+Example:
+
+```python
+In [7]: df1 = pd.DataFrame({'a': [1,  2], 'b': [4, 5]}, index=NoRowIndex(2))
+
+In [8]: df2 = pd.DataFrame({'a': [4], 'b': [0]}, index=NoRowIndex(1))
+
+In [9]: df1
+Out[9]:
+ a  b
+ 1  4
+ 2  5
+
+In [10]: pd.concat([df1, df2])
+Out[10]:
+ a  b
+ 1  4
+ 2  5
+ 4  0
+
+In [11]: pd.concat([df1, df2]).index
+Out[11]: NoRowIndex(len=3)
+```
+
+Appending anything other than another ``NoRowIndex`` would raise.
+
+### Slicing a ``NoRowIndex``
+
+If one has a ``DataFrame`` with ``NoRowIndex``, then one would expect that a slice of it would still have
+a ``NoRowIndex``. This could be accomplished with:
+
+> If a slice of length ``x`` is taken from a ``NoRowIndex`` of length ``y``, then one gets a
+ ``NoRowIndex`` of length ``x``. Label-based slicing would not be allowed.
+
+Example:
+
+```python
+In [12]: df = pd.DataFrame({'a': [1,  2, 3], 'b': [4, 5, 6]}, index=NoRowIndex(3))
+
+In [13]: df.loc[df['a']>1, 'b']
+Out[13]:
+5
+6
+Name: b, dtype: int64
+
+In [14]: df.loc[df['a']>1, 'b'].index
+Out[14]: NoRowIndex(len=2)
+```
+
+Slicing by label, however, would be disallowed:
+```python
+In [15]: df.loc[0, 'b']
+---------------------------------------------------------------------------
+IndexError: Cannot use label-based indexing on NoRowIndex!
+```
+Note that other uses of ``.loc``, such as boolean masks, would still be allowed (see F.A.Q).
```

Comment:
Do you need to say that `.iloc` would work just fine, that `DataFrame.at` would not work, but `DataFrame.iat` would work?

Reply:
Yup, will do, thanks for your comments!

On contractions, MLA allow them, noting that they can "help make text more approachable and easier to read" https://style.mla.org/contractions/
Can remove them if you have strong feelings on them, just noting that they are becoming more accepted in writing

Reply:
> On contractions, MLA allow them, noting that they can "help make text more approachable and easier to read" https://style.mla.org/contractions/
> Can remove them if you have strong feelings on them, just noting that they are becoming more accepted in writing

To me, a PDEP is a type of scientific writing, and in the 30+ papers I have written, I haven't used contractions (although I just used one in the previous sentence, because this is an informal discussion).  So by removing them, it makes the document more formal, as opposed to an informal discussion in a comment.  Would be curious to see what others think.

Reply:
Contractions are used on peps (see peps.python.org), but if you're trying to declare a hard rule, I agree, better to use well defined rfc type language.